src/pat/masonry/masonry.js

```diff
@@ -12,51 +12,54 @@
             "pat-parser",
             "pat-base",
             "pat-utils",
-            "masonry",
-            "imagesloaded"
+            "masonry"
             ], function() {
                 return factory.apply(this, arguments);
         });
     } else {
-        factory(root.$, root.patterns, root.patterns.Parser, root.Base, root.Masonry, root.imagesLoaded);
+        factory(root.$, root.patterns, root.patterns.Parser, root.Base, root.Masonry);
     }
-}(this, function($, logger, registry, Parser, Base, utils, Masonry, imagesLoaded) {
+}(this, function($, logger, registry, Parser, Base, utils, Masonry) {
     "use strict";
     var log = logger.getLogger("pat.masonry");
     var parser = new Parser("masonry");
+
+    // parser.addArgument("stagger", "");
     parser.addArgument("column-width");
     parser.addArgument("container-style", '{ "position": "relative" }');
     parser.addArgument("gutter");
-    parser.addArgument("hidden-style", "{ opacity: 0, transform: 'scale(0.001)' }");
     parser.addArgument("is-fit-width", false);
+    parser.addArgument("is-horizontal-order", false);  // preserve horizontal order.
     parser.addArgument("is-origin-left", true);
     parser.addArgument("is-origin-top", true);
+    parser.addArgument("is-percent-position", false);  // set item positions in percent values. items will not transition on resize.
+    parser.addArgument("is-resize", true);  // adjust sizes and position on resize.
     parser.addArgument("item-selector", ".item");
     parser.addArgument("stamp", "");
     parser.addArgument("transition-duration", "0.4s");
-    parser.addArgument("visible-style", "{ opacity: 1, transform: 'scale(1)' }");
+
+    // is-* are masonry v3 options, here we add v4 style names.
+    // we keep the is-* as there is special support with options parsing.
+    parser.addAlias("fit-width", "is-fit-width");
+    parser.addAlias("origin-left", "is-origin-left");
+    parser.addAlias("origin-top", "is-origin-top");
+    parser.addAlias("horizontal-order", "is-horizontal-order");
+    parser.addAlias("percent-position", "is-percent-position");
+    parser.addAlias("resize", "is-resize");
+
+
 
     return Base.extend({
         name: "masonry",
         trigger: ".pat-masonry",
 
         init: function masonryInit($el, opts) {
             this.options = parser.parse(this.$el, opts);
-            var imgLoad = imagesLoaded(this.$el);
-            imgLoad.on("progress", function() {
-                if (! this.msnry) {
-                    this.initMasonry();
-                }
-                this.quicklayout();
-            }.bind(this));
-            imgLoad.on("always", function () {
-                if (! this.msnry) {
-                    this.initMasonry();
-                }
-                this.layout();
-            }.bind(this));
```

Comment:
Hi @thet, can you tell me how you replace the imagesloaded functionality? This commit looks like you basically replaced that with the initMasonry call only. But the layout call is missing, so my masonries are not rendering. How are you doing this?

Reply:
I don't yet understand how the imagesloaded part is being replaced. I see timing issues again where the masonry is layouting while not all images are loaded yet. That results in views like the attached screenshot. Until we find out, I am tempted to readd the imagesloaded code.
![bildschirmfoto 2018-06-17 um 16 56 48](https://user-images.githubusercontent.com/138906/41509133-84e59ec8-724f-11e8-8f30-53e6df5eae4b.png)

Reply:
The replacement for imagesloaded is in the following lines in the 
https://github.com/Patternslib/Patterns/pull/520 PR for the 2.1.x branch. Especially the "load" event handler on image elements:
```
            // Initially layout on document ready.
            $(document).ready(utils.debounce(this.layout.bind(this), 100));

            // Re-Layout, if images are loaded within pat-masonry
            $('img', this.$el)
.on("load", utils.debounce(this.quicklayout.bind(this), 200));
```

That got lost in this PR. Here an observer is used.

Reply:
@pilz does this look only when the site is loaded or even after all images have been loaded?

Reply:
The observer is looking for changes in the DOM only. That one addresses issues when something is injected outside the masonry context so that masonry can adapt to the new space situation and rerender. images loaded is not addressed by the observer. Can you give me some background why the img.load event handler is better? Is that something new? Why does imagesLoaded exist if you can do this that easy? And did imagesLoaded give you some problems so that you had to replace it with the on load handler?

Reply:
@pilz I had some issues with image loading and masonry resizing. It did look quite ugly - all images were displayed on top of each other until they were finally loaded. The optimization fixed that somehow. I saw the ``load`` event making imagesloaded obsolete - having fewer dependencies is also a nice thing.
I remember to have seen some docs hinting that the load event finally works in modern browsers and does not suffer from the issues mentioned here: http://api.jquery.com/load/ - but I can't find it yet and that might be a wrong memory. When I find out more, I'll let you know....